servconf.c

```diff
@@ -223,6 +223,10 @@ assemble_algorithms(ServerOptions *o)
 	def_kex = match_filter_allowlist(KEX_SERVER_KEX, all_kex);
 	def_key = match_filter_allowlist(KEX_DEFAULT_PK_ALG, all_key);
 	def_sig = match_filter_allowlist(SSH_ALLOWED_CA_SIGALGS, all_sig);
+	if (NULL == def_key || NULL == def_sig) { // fix CodeQL SM02311
```

Comment:
The codeql issue seems to be complaining about `all_key` and` all_sig` being null, not `def_key` and `def_sig`. Should we be checking those instead?